logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserModel.kt

```diff
@@ -2,13 +2,22 @@ package com.wire.kalium.logic.data.user
 
 import com.wire.kalium.logic.data.id.ConversationId
 import com.wire.kalium.logic.data.id.QualifiedID
+import com.wire.kalium.logic.data.id.TeamId
 import com.wire.kalium.logic.data.id.VALUE_DOMAIN_SEPARATOR
-import com.wire.kalium.logic.data.publicuser.model.OtherUser
+import com.wire.kalium.logic.data.user.type.UserType
 
 typealias UserId = QualifiedID
 
-abstract class User {
+sealed class User {
     abstract val id: UserId
+    abstract val name: String?
+    abstract val handle: String?
+    abstract val email: String?
+    abstract val phone: String?
+    abstract val accentId: Int
+    abstract val teamId: TeamId?
+    abstract val previewPicture: UserAssetId?
+    abstract val completePicture: UserAssetId?
 }
```

Comment:
why sealed class ?

Reply:
it's just so the compiler can warn us about the missing branches when we use the `when` keyword